File: src/main/java/org/noxet/noxetserver/messaging/ErrorMessage.java
package org.noxet.noxetserver.messaging;

import net.md_5.bungee.api.ChatColor;

public class ErrorMessage extends Message {
    public enum ErrorType {
        COMMON(null),
        PERMISSION("missing permission"),
        ARGUMENT("incorrect argument usage");

        private final String typeString;

        ErrorType(String typeString) {
            this.typeString = typeString;
        }

        public String getTypeString() {
            return typeString;
        }
    }

    private final ErrorType errorType;

    public ErrorMessage(ErrorType errorType, String text) {
        super(text);
        this.errorType = errorType;
    }

    @Override
    public ChatColor getDefaultColor() {
        return ChatColor.RED;
    }

    @Override
    public String getPrefix() {
        return super.getPrefix() + "§4Error" + (errorType.getTypeString() != null ? " (§7" + errorType.getTypeString() + "§4)" : "") + ": ";
    }
}

File: src/main/java/org/noxet/noxetserver/messaging/ErrorMessage.scala
package org.noxet.noxetserver.messaging

import net.md_5.bungee.api.ChatColor

class ErrorMessage(kind: ErrorMessage.ErrorType, text: String) extends Message(text):
    override def getDefaultColor: ChatColor = ChatColor.RED
    override def getPrefix: String =
        super.getPrefix + "§4Error" + kind.desc.map(d => s" (§7$d§4)").getOrElse("") + ": "
        
object ErrorMessage:
    enum ErrorType(val desc: Option[String]):
        case Common extends ErrorType(None)
        case Permission extends ErrorType(Some("missing permission"))
        case Argument extends ErrorType(Some("incorrect argument usage"))